platform_umbrella/apps/common_core/lib/common_core/resources/postgres/cloudnative_pg_clusters.ex

```diff
@@ -151,6 +153,7 @@ defmodule CommonCore.Resources.CloudnativePGClusters do
     |> B.spec(spec)
     |> F.require_non_nil(battery.config.bucket_name)
     |> F.require_non_nil(battery.config.service_role_arn)
+    |> F.require(cluster.backup_config.type == :object_store)
```

Comment:
Remember that backup_config might be nil.
```suggestion
    |> F.require(cluster.backup_config != nil && cluster.backup_config.type == :object_store)
```